### PR TITLE
[P8-49] ATL formula parser

### PR DESCRIPTION
**Pull request by jenrik**
Parses ATL formula into a `Phi` data structure

### CODE REVIEW

File: src/atl/formula/parser.rs
Comment (NicEastvillage) [inline src/atl/formula/parser.rs:331]: I am not sure this should be allowed. Same goes for the other operators. But it's a small thing.

**Review by NicEastvillage — APPROVED** [review]
Nice

**Comment by jenrik** [comment]
The `convert` function needs to be split into a function for converting players and a function for propositions.

**Comment by NicEastvillage** [comment]
Yeah, I think we should introduce a trait with two functions: `parse_player` and `parse_proposition`

**Comment by NicEastvillage** [comment]
This feature is done and waiting for reviews from @d702e20/d702e20

File: src/atl/formula/parser.rs
Comment (jenrik) [inline src/atl/formula/parser.rs:14]: We should allow whitespace at the end of a formula like we do at the beginning
```suggestion
    let formula = ws() * phi(expr_parser) - ws() - end();
```

File: src/atl/formula/parser.rs
Comment (jenrik) [inline src/atl/formula/parser.rs:27]: This should return `Parser<u8, crate::common::Proposition>`. `Proposition` is a alias for `usize` but having a use-case specific name for it helps if we ever need to refactor.

File: src/lcgs/ir/intermediate.rs
Comment (jenrik) [inline src/lcgs/ir/intermediate.rs:522]: Why are these lines in parenthesis?

Comment (NicEastvillage) [reply src/lcgs/ir/intermediate.rs:522]: Becase the block returns a tuple consisting of a SymbolIdentifier and a String. See line 517 - it's a let statement.

File: src/atl/formula/parser.rs
Comment (NicEastvillage) [reply src/atl/formula/parser.rs]: Good spot. Fixed in 0d67118

File: src/atl/formula/parser.rs
Comment (NicEastvillage) [reply src/atl/formula/parser.rs]: Good idea. Fixed in 0d67118

Comment (jenrik) [reply src/atl/formula/parser.rs]: Now that I think about, `player_parser` should probably return `crate::common::Player`

**Review by lars4509 — APPROVED** [review]
Looks fine to me, i just have the single comment.

File: src/main.rs
Comment (jenrik) [inline src/main.rs:262]: Unneeded `return` statement

**Review by jenrik — COMMENTED** [review]
Highlighted some Clippy warnings

File: docs/atl-grammar.txt
Comment (MathiasMehl) [reply docs/atl-grammar.txt:41]: I see your point, but I am in a different mindset when writing ATL formulae as opposed to LCGS, so I personally don't confuse the syntax. 
I'm fine with either solution (changing or not).

Comment (jenrik) [reply docs/atl-grammar.txt:41]: I would rather that we change LCGS, since the formula syntax is based on a previously established syntax for LTS formulae that is also used elsewhere

Comment (MathiasMehl) [reply docs/atl-grammar.txt:41]: I assume you mean LTL? AFAIK LTL uses  ∧

Comment (NicEastvillage) [reply docs/atl-grammar.txt:41]: Depends on if we ever want to support bitwise operators in LCGS.

**Review by lars4509 — APPROVED** [review]

**Review by MathiasMehl — APPROVED** [review]

**Pull request merged**
